x-pack/plugin/logsdb/src/yamlRestTest/resources/rest-api-spec/test/40_source_mode_setting.yml

```diff
@@ -0,0 +1,833 @@
+        index: test_unset_stored
+        body:
+          settings:
+            index:
+              mapping.source.mode: stored
+
+  - do:
+      indices.get_mapping:
+        index: test_unset_stored
+
+  - match: { test_unset_stored.mappings: { } }
+
+---
+override unspecified to disabled source mode:
+  - do:
+      indices.create:
+        index: test_unset_disabled
+        body:
+          settings:
+            index:
+              mapping.source.mode: disabled
+
+  - do:
+      indices.get_mapping:
+        index: test_unset_disabled
+
+  - match: { test_unset_disabled.mappings: { } }
+
+---
+override unspecified to synthetic source mode:
+  - do:
+      indices.create:
+        index: test_unset_synthetic
+        body:
+          settings:
+            index:
+              mapping.source.mode: synthetic
+
+  - do:
+      indices.get_mapping:
+        index: test_unset_synthetic
+
+  - match: { test_unset_synthetic.mappings: { } }
+
+---
+create an index with standard index mode:
+  - do:
+      indices.create:
+        index: test_standard_index_mode
+        body:
+          settings:
+            index:
+              mode: standard
+          mappings:
+            _source:
+              mode: stored
+
+  - do:
+      indices.get_mapping:
+        index: test_standard_index_mode
+
+  - match: { test_standard_index_mode.mappings._source.mode: stored }
+
+---
+create an index with time_series index mode and synthetic source:
+  - do:
+      indices.create:
+        index: test_time_series_index_mode_synthetic
+        body:
+          settings:
+            index:
+              mode: time_series
+              mapping.source.mode: synthetic
+              routing_path: [ keyword ]
+              time_series:
+                start_time: 2021-04-28T00:00:00Z
+                end_time: 2021-04-29T00:00:00Z
+          mappings:
+            properties:
+              keyword:
+                type: keyword
+                time_series_dimension: true
+
+  - do:
+      indices.get_settings:
+        index: "test_time_series_index_mode_synthetic"
+  - match: { test_time_series_index_mode_synthetic.settings.index.mode: time_series }
+
+
+  - do:
+      indices.get_mapping:
+        index: test_time_series_index_mode_synthetic
+
+  - match: { test_time_series_index_mode_synthetic.mappings._source.mode: synthetic }
+
+---
+create an index with logsdb index mode and synthetic source:
+  - do:
+      indices.create:
+        index: test_logsdb_index_mode_synthetic
+        body:
+          settings:
+            index:
+              mode: logsdb
+              mapping.source.mode: synthetic
+
+  - do:
+      indices.get_settings:
+        index: "test_logsdb_index_mode_synthetic"
+  - match: { test_logsdb_index_mode_synthetic.settings.index.mode: logsdb }
+
+  - do:
+      indices.get_mapping:
+        index: test_logsdb_index_mode_synthetic
+
+  - match: { test_logsdb_index_mode_synthetic.mappings._source.mode: synthetic }
+
+#---
```

Comment:
@lkts @martijnvg I had to comment out htese two tests failing in the *assert* here

```
// synchronized concurrent mapper updates are guaranteed to set merged mappers derived from the mapper value previously read
            // TODO: can we even have concurrent updates here?
            synchronized (this) {
                Mapping mapping = mergeMappings(this.mapper, incomingMapping, reason, this.indexSettings);
                DocumentMapper newMapper = newDocumentMapper(mapping, reason, mapping.toCompressedXContent());
                this.mapper = newMapper;
                assert assertSerialization(newMapper, reason);
                return newMapper;
            }
```

it looks like we have a condition where we try to change the source mode from synthetic to stored.
The other PR https://github.com/elastic/elasticsearch/pull/114454 changed the behavior of this test and we are not allowed to have conflicting source modes using both an index mode and a setting. The setting take precedence and determines that the source mode is stored. At merging time there is a conflict between synthetic and stored source mode which makes the assertion fail.